cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapperWrapper.java

```diff
@@ -156,6 +178,16 @@ public void run(Context context) throws IOException, InterruptedException {
 
       @Override
       public boolean nextKeyValue() throws IOException, InterruptedException {
+        // If entity id is not null, verify that the user has read access to the entity.
+        if (entityId != null) {
+          try {
+            authorizationEnforcer.enforce(entityId, authenticationContext.getPrincipal(), Action.READ);
+          } catch (Exception e) {
+            Throwables.propagateIfInstanceOf(e, IOException.class);
+            throw new IOException(e);
```

Comment:
so if there's an ``UnauthorizedException`` here, it will be propagated as an ``IOException``? Hmmm, makes me think if we should make ``UnauthorizedException`` unchecked, and just throw as-is. Unrelated to this PR, but what do you think?